app/src/main/java/ch/epfl/cs311/wanderwave/viewmodel/BeaconViewModel.kt

```diff
@@ -0,0 +1,60 @@
+package ch.epfl.cs311.wanderwave.viewmodel
+
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import ch.epfl.cs311.wanderwave.model.data.Beacon
+import ch.epfl.cs311.wanderwave.model.data.Location
+import ch.epfl.cs311.wanderwave.model.data.Profile
+import ch.epfl.cs311.wanderwave.model.data.ProfileTrackAssociation
+import ch.epfl.cs311.wanderwave.model.data.Track
+import ch.epfl.cs311.wanderwave.model.repository.BeaconRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class BeaconViewModel @Inject constructor(private val beaconRepository: BeaconRepository) :
+    ViewModel() {
+
+  private var _uiState = MutableStateFlow(UIState())
+
+  val uiState: StateFlow<UIState> = _uiState
+
+  init {
+    val sampleBeacon =
```

Comment:
I'm not sure this is the absolute best way to do it, maybe we should rather have the beacon start as null, and add some logic to handle this scenario instead of creating a placeholder beacon. Let me know what you think

Reply:
This is for the testing, this is not him that did it. I will change it later but for not it's just like that :(